index.html

```diff
@@ -2406,6 +2634,1425 @@ <h2>
+          </p>
+          <table>
+            <tr>
+              <th>
+                Action
+              </th>
+              <th>
+                Non-normative Description
+              </th>
+            </tr>
+            <tr>
+              <td>
+                GamepadAxisInput
+              </td>
+              <td>
+                Used to indicate that the [=input value=] for an [=axis=] has updated.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                GamepadButtonInput
+              </td>
+              <td>
+                Used to indicate that the [=input values=] for a [=button=] have updated.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                GamepadTouchStartAction
+              </td>
+              <td>
+                Used to indicate that there is a new [=touch point=] on a [=touch surface=].
+              </td>
+            </tr>
+            <tr>
+              <td>
+                GamepadTouchMoveAction
+              </td>
+              <td>
+                Used to indicate that an active [=touch point=] has updated coordinates.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                GamepadTouchEndAction
+              </td>
+              <td>
+                Used to indicate that a [=touch point=] is no longer active.
+              </td>
+            </tr>
+          </table>
+          <p>
+            To <dfn>create a gamepad input source</dfn> object given |params|, return a new [=gamepad input source=] with:
+          </p>
+          <ul>
+            <li>
+              `axes` set to |params|.axes,
+            </li>
+            <li>
+              `buttons` set to |params|.buttons,
+            </li>
+            <li>
+              `surfaces` set to |params|.surfaces, and
+            </li>
+            <li>
+              `vibration` set to |params|.vibration.
+            </li>
+          </ul>
+        </section>
+        <section>
+          <h4>
+            Actions
+          </h4>
+          <p>
+            To <dfn>process a gamepad action</dfn> given |id| and |action item|:
+          </p>
+          <ol>
+            <li>
+              Let |subtype| be the result of getting the property named "type" from |action item|.
+            </li>
+            <li>
+              If |subtype| is not one of the values `"axis"`, `"button"`, `"touchStart"`, `"touchMove"`, `"touchEnd"`, or `"pause"`, return an [=error=] with [=error code=] [=invalid argument=].
+            </li>
+            <li>
+              Let |action| be an [=action object=] constructed with arguments |id|, `"gamepad"`, and |subtype|.
+            </li>
+            <li>
+              If |subtype| is `"pause"`, let |result| be the result of [=trying=] to [=process a pause action=] with arguments |action item| and |action|, and return |result|.
```

Comment:
This and other calls can be simplified:
```suggestion
              If |subtype| is `"pause"`, return the result of [=trying=] to [=process a pause action=] |action item| and |action|.
```